Deprecation Warning on "._headers"

Hello,

The package seems to work as intended, but produces the following deprecation warning:

```
(node:12797) [DEP0066] DeprecationWarning: OutgoingMessage.prototype._headers is deprecated
```

For reference, node provides the [following explanation](https://nodejs.org/api/deprecations.html#deprecations_dep0066_outgoingmessage_prototype_headers_outgoingmessage_prototype_headernames) on the matter.

I attempted to fix it myself using node's recommended public method, but can't get the project to compile as it produces the following errors.

```
> express-idempotency@1.0.4 compile
> ./node_modules/.bin/tsc --build tsconfig.dist.json

node_modules/@types/express-serve-static-core/index.d.ts:589:18 - error TS2430: Interface 'Response<ResBody, Locals, StatusCode>' incorrectly extends interface 'ServerResponse'.
  Property 'req' is optional in type 'Response<ResBody, Locals, StatusCode>' but required in type 'ServerResponse'.

589 export interface Response<
                     ~~~~~~~~

node_modules/@types/express/ts4.0/index.d.ts:43:55 - error TS2344: Type 'Response<any, Record<string, any>>' does not satisfy the constraint 'ServerResponse'.
  Property 'req' is optional in type 'Response<any, Record<string, any>>' but required in type 'ServerResponse'.

43     var static: serveStatic.RequestHandlerConstructor<Response>;
                                                         ~~~~~~~~


Found 2 errors.
```

To get the following messages, I cloned the repo, ran `npm install` and then `npm run compile`. I'll will probably look further at the cause of those errors at some point later if you don't have the time to do so, but I expect your knowledge of the project might speed up resolving this issue.

--- a/examples/simple-use/Dockerfile
+++ b/examples/simple-use/Dockerfile
@@ -1,4 +1,4 @@
-FROM node:10
+FROM node:16
 
 # Create app directory
 WORKDIR /usr/src/app
@@ -7,16 +7,18 @@ WORKDIR /usr/src/app
 COPY . .
 
 # Compile the library and link the library
-RUN npm install && npm run compile && npm prune --production && npm link
+RUN rm -rf node_modules dist && npm install && npm run compile && npm prune --production && npm link
 
 # Installation of package to run the application
 WORKDIR /usr/src/app/examples/simple-use
 
-RUN npm link "express-idempotency" && npm install --production
+RUN rm -rf node_modules && npm link "express-idempotency" && npm install --production
+# Force use of local library, npm link is not always working for some reasons
+RUN npm install ../../.
 
 EXPOSE 8080
 
 # Run the application
 WORKDIR /usr/src/app
-CMD ["node", "./examples/simple-use/src/index.js"]
+CMD ["node", "--trace-deprecation", "./examples/simple-use/src/index.js"]
 

--- a/examples/simple-use/src/index.js
+++ b/examples/simple-use/src/index.js
@@ -16,7 +16,9 @@ app.all('/', function (req, res) {
     }
 
     // Return the current date and time
-    res.send(new Date().toISOString());
+    res.json({
+        value: new Date().toISOString()
+    });
 });
 
 app.listen(8080, function () {

--- a/src/services/idempotencyService.ts
+++ b/src/services/idempotencyService.ts
@@ -22,11 +22,6 @@ const IDEMPOTENCY_KEY_HEADER = 'idempotency-key';
 const HIT_HEADER = 'x-hit';
 const HIT_VALUE = 'true';
 
-// tslint:disable-next-line:interface-name
-interface InternalExpressResponse extends express.Response {
-    _headers: any;
-}
-
 /**
  * This class represent the idempotency service.
  * It contains all the logic.

--- a/src/services/idempotencyService.ts
+++ b/src/services/idempotencyService.ts
@@ -293,7 +293,7 @@ export class IdempotencyService {
     ): IdempotencyResponse {
         const headerWhitelist: string[] = ['content-type'];
         const preliminaryHeaders = res.getHeaders();
-        
+
         // Keeps only whitelisted headers
         const headers = Object.keys(preliminaryHeaders)
             .filter((key) => headerWhitelist.includes(key))

--- a/src/services/idempotencyService.ts
+++ b/src/services/idempotencyService.ts
@@ -292,7 +292,8 @@ export class IdempotencyService {
         body: any
     ): IdempotencyResponse {
         const headerWhitelist: string[] = ['content-type'];
-        const preliminaryHeaders = (res as InternalExpressResponse)._headers;
+        const preliminaryHeaders = res.getHeaders();
+        
         // Keeps only whitelisted headers
         const headers = Object.keys(preliminaryHeaders)
             .filter((key) => headerWhitelist.includes(key))

